bigchaindb/core.py

```diff
@@ -343,6 +344,21 @@ def get_txs_by_asset_id(self, asset_id):
         cursor = self.backend.get_transactions_by_asset_id(asset_id)
         return [Transaction.from_dict(tx) for tx in cursor]
 
+    def get_asset_by_id(self, asset_id):
+        """Returns the asset associated with an asset_id
+
+            Args:
+                asset_id (str): The asset id
+
+            Returns:
+                :class:`~bigchaindb.common.transaction.Asset` if the asset
+                exists else None
+        """
+        cursor = self.backend.get_asset_by_id(asset_id)
+        cursor = list(cursor)
+        if cursor:
+            return Asset.from_dict(cursor[0]['transaction']['asset'])
```

Comment:
Exactly, #788 will address this. The idea is that the DB layer handles raw documents. Once the document is loaded in the `core.py::Bigchain` class it will be imported in a model.